### DISCUSSION TITLE
Keyword Auto Refresh Question

### DISCUSSION THREAD

**Discussion by jbk68** [cortexrd/Knack-Toolkit-Library#155] [General]
Out of all the great solutions KTL offers, one of my favorite is the keyword for auto refreshing (_ar) a particular view.  But one issue that I would like to see if there is a solution for is the following.  Using the below code, we have changed the wording on a search field from "search by keyword" to "search by serial number".  The problem is, when a user initially opens this particular view, the wording on the search bar is "search by serial number" but because we have the auto refreshing (_ar) set to "quietly" refresh the page every 10 seconds, that the "search by keyword" will replace our custom wording.  If user leaves page and comes back to it or if they refresh browser, our custom wording will reappear in the search bar but within 10 seconds be replaced by generic built in wording of "search by keyword"

I'd like to see if there is a way to keep our custom search bar wording regardless of using KTL's auto refresh feature.

Here is the code I used to replace Knacks generic wording in search bar:

$(document).on('knack-scene-render.any', function(event, scene) {
  $("input[name='keyword']").attr("placeholder", "search by serial number")
});

Thank you... JON

**Comment by amandajane-mo** [comment]
Hi Jon,

Have you tried changing your code to run on any view?

```
$(document).on('knack-view-render.any', function(event, view) {
   $("input[name='keyword']").attr("placeholder", "search by serial number");
});
```

Amanda

**Reply by jbk68** [reply]
Hey Amanda, thank you for the quick response and more importantly, **a working solution.**  

Thanks again... JON

**Reply by jbk68** [reply]
Hey Amanda, question for you.  I have to grids on the same scene and I want both of the them to have different wording in the search bar.

View 95 to have generic phrase of "search by keyword"   **and**  View 233 to have "search by serial number"

I used the below code to change view 233's search bar phrase but it's changing view 95's to "search by serial number" upon refreshing view using _ar.   I have each view set to auto refresh every 10 seconds.

$(document).on('knack-view-render.view_233', function(event, view) {
   $("input[name='keyword']").attr("placeholder", "search by serial number");
});

Thank you... JON

**Comment by amandajane-mo** [comment]
Hi Jon,

You just need to add the view ID to your jquery selector, this will prevent any other views from being effected:

```
$(document).on('knack-view-render.view_233', function(event, view) {
$(`#${view.key} input[name='keyword']`).attr("placeholder", "search by serial number");
});
```

Do you do this a lot? If so, you could turn it into a function, that excepts the view.key and the placeholder text and then call it where needed. 

Amanda

**Reply by jbk68** [reply]
Thanks Amanda that worked.  To answer your question, this is the only time I use this... JON
